standards/request-response.md

```diff
@@ -493,6 +493,36 @@ The usage of non-standard headers is not considered custom headers. For example,
 SPS-CORS-Error: bad origin
 ```
 
+#### SPS-Execution-Context
+
+**Type**: Request
+
+**Support**: OPTIONAL
+
+**Description**: The Execution Context header provides a standard method of indicating the dataflow context that a request should be processed with. This is used to differentiate or test changes in customer configuration of non-production data. This isolates the version of the configuration state used in a request and allows identifying transactions that should leverage a specific tagged or aliased configuration set.
+
+- A request without the presence of this header **MUST** indicate that the execution context is under the `production` dataflow and customer configuration, if applicable.
+- The header value contains any dynamic __string__ value representing a named configuration or tagged dataflow preset.
+    - __preprod__ - Well-known value used to represent non-production customer configuration where named configuration is not supported. Typically used for legacy support. Usage of dynamically named configuration sets is preferred.
+- An invalid or unsupported header value provided **MUST** result in a `400 - Bad Request` following standard [error format](errors.md#400-bad-request).
+- The header value **MUST NOT** exceed a maximum length of 100 characters.
+- The header value **SHOULD** contain human-readable tag for the context.
+- The header **MUST NOT** be used in a response.
```

Comment:
Why? It might be useful to return back (especially if it's async) to caller so the integrity of the context is maintained. That said, it should be preserved in response only if it was as part of the original request.

Reply:
I agree with this - in fact we had this discussion and you wanted as request only. But yes if we do modify this:
- What is the response if none in the request?
- Is this response header always given? (I think it should be)

Reply:
What is the response if none in the request?... I'd say optional unless it's by-design of this API to return it. Maybe in that situation, it's not even a header, but a response body. Like API that returns available contexts.

Is this response header always given? (I think it should be) ... as long as it's provided in the request, otherwise, it's optional to return based on the behavior of API. (if I understood question right)

Reply:
So to clarify: The header SHOULD be returned as a response header if it was provided in the request?

- Why not make it a "MUST BE RETURNED" if in the request for consistency?
- Is there a scenario where a service may want to return the response header for every request, regardless of if its present in the request? If so, does that imply we need a value for "PRODUCTION" to rally around (like "PREPROD")?

Reply:
1 - makes sense to me
2 - that's something we touched on earlier and likely a more long-term view of this. PRODUCTION essentially can be equal to EMPTY. Might be something worth discussing a bit more, but I don't see reason why not.

Reply:
Well I think it would be odd to have a header with no value.

So what we are saying is if the header is supplied on the request, then the response returns it. If it is not supplied, it does not return it. Based on this, you would never expect to see the request or response header provided that included the term (or similar term) "production"?